tests/integration/bond_test.yaml

```diff
@@ -0,0 +1,375 @@
+description: Delete the bond device bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: absent
+description: Add bond interface bond99 with bonding mode active-backup, attach two ports eth1 and eth2, configure the port eth1 with priority -1 and queue ID 0, configure the port eth2 with priority 2 and queue ID 1
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: active-backup
+    ports-config:
+    - name: eth1
+      priority: -1
+      queue-id: 0
+    - name: eth2
+      priority: 2
+      queue-id: 1
+description: Add bond interface bond99 with two ports eth1 and eth2, set bonding mode balance-rr
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-rr
+    port:
+    - eth1
+    - eth2
+description: Remove bonding interface bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: absent
+description: Remove bond interface bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: absent
+description: Add the bond interface bond99, configure the static IP 192.168.122.250 with prefix length 24 for bond99, set the bonding mode to balance-rr, attach eth1 and eth2 as the ports, set the bond options miimon 140
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  ipv4:
+    enabled: true
+    address:
+    - ip: 192.168.122.250
+      prefix-length: 24
+  link-aggregation:
+    mode: balance-rr
+    options:
+      miimon: 140
+    port:
+    - eth1
+    - eth2
+description: Configure bond interface bond99 with the port eth1 and eth2, configure the port eth1 with priority -1 and queue ID 1, configure the port eth2 with priority 9 and queue ID 0
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    ports-config:
+    - name: eth1
+      priority: -1
+      queue-id: 1
+    - name: eth2
+      priority: 9
+      queue-id: 0
+description: Delete bond interface bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: absent
+description: Add bond interface bond99 with bonding mode active-backup, attach two ports eth1 and eth2. Configure eth1 with the bond port priority -1 and queue id 0, configure eth2 with bond port priority 2 and queue id 1
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: active-backup
+    ports-config:
+    - name: eth1
+      priority: -1
+      queue-id: 0
+    - name: eth2
+      priority: 2
+      queue-id: 1
+description: Create bond interface bond99 with port eth1 attached
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    port:
+    - eth1
+description: Attach eth1 and eth2 as ports to bond99
```

Comment:
This description sounds like bond99 has other ports.

`Attach xxx to bond99` should be used to describe `controller` property.

Reply:
how about `Configure bond99 with ports eth1 and eth2` ? I was using `attach xxx` because I was thinking that this is modifying the state.